src/components/App/App.js

```diff
@@ -0,0 +1,359 @@
+    mainApi.register(email, password, username)
+      .then((res) => {
+        if (res) {
+          handleSignupSuccessClick();
+          console.log("Register success");
+          // resetForm();
+          // setEmailConflict(false);
+        }
+      })
+      .catch((err) => {
+        // if (err.includes(409)) {
+        //   // setEmailConflict(true);
+        // }
+        console.log("Something went wrong.");
+      });
+  }
+
+  function handleSubmitLogin(email, password, resetForm) {
+    mainApi.login(email, password)
+      .then((data) => {
+        if (data.token) {
+          setToken(data.token);
+          setLoggedIn(true);
+          closeAllPopups();
+          // history.push("/");
+          resetForm();
+        }
+      })
+      .catch((err) => console.log(err));
+  }
+
+  function handleLogOutClick() {
+    localStorage.removeItem("jwt");
+    setLoggedIn(false);
+    closeAllPopups();
+  }
+
+  function closeAllPopups() {
+    setIsSignInPopupOpen(false);
+    setIsSignUpPopupOpen(false);
+    setSignupSuccessOpen(false);
+    setNewsCardBtnHover(false);
+    // setIsNavMobileOpen(false);
+  }
+
+  React.useEffect(() => {
+    const closeByEscape = (e) => {
+      if (e.key === "Escape") {
+        closeAllPopups();
+      }
+    };
+    document.addEventListener("keydown", closeByEscape);
+    return () => document.removeEventListener("keydown", closeByEscape);
+  }, []);
+
+  React.useEffect(() => {
+    if (localStorage.getItem("keyword")) {
+      setKeyword(localStorage.getItem("keyword"));
+    }
+  }, []);
+
+  return (
+    <div className='app'>
+      <CurrentUserContext.Provider value={currentUser}>
+        <div className="app__container">
+          <Header
+            onSignInClick={handleSignInClick}
+            onLogOutClick={handleLogOutClick}
+            onOpenNavClick={handleOpenNavClick}
+            isNavOpen={isNavMobileOpen}
+            loggedIn={loggedIn}
+            onNavClose={handleSavedArticlesClick}
+
+          />
+          <Routes>
+            <Route path='/' element={<Home
+              isHover={isNewsCardBtnHover}
+              onNewsCardBtnHover={handleNewsCardBtnHover}
+              onNewsCardBtnClose={closeAllPopups}
+              onSearchSubmit={handleSearchSubmit}
+              keyword={keyword}
+              handleChange={handleKeywordChange}
+              cards={cards}
+              isArticlesOpen={isArticlesOpen}
+              isLoaderOpen={isLoaderOpen}
+              isNothingFoundOpen={isNothingFoundOpen}
+              isErrMessageOpen={isErrMessageOpen}
+              addArticle={handleAddArticle}
+              handleSaveCardClick={handleSaveCardClick}
+            />} />
+            {/* <Route  path='/'>
+            <Home />
+          </Route> */}
+            <Route path='/saved-articles' element={
+              <ProtectedRoute loggedIn={loggedIn}>
+                <SavedArticlesContext.Provider value={savedArticles}>
```

Comment:
`                <SavedArticlesContext.Provider value={savedArticles}>
`

`SavedArticlesContext` is not used inside components so you don't need this context then.


You pass the articles into the props